src/flyte/_run.py

```diff
@@ -462,15 +466,26 @@ async def _run_local(self, obj: TaskTemplate[P, R], *args: P.args, **kwargs: P.k
             compiled_image_cache=None,
             report=Report(name=action.name),
             mode="local",
+            input_context=self._input_context,
         )
-        with ctx.replace_task_context(tctx):
-            # make the local version always runs on a different thread, returns a wrapped future.
-            if obj._call_as_synchronous:
-                fut = controller.submit_sync(obj, *args, **kwargs)
-                awaitable = asyncio.wrap_future(fut)
-                outputs = await awaitable
-            else:
-                outputs = await controller.submit(obj, *args, **kwargs)
+
+        # Set input context from with_runcontext if provided
+        context_token = None
+        if self._input_context:
+            context_token = _input_context_var.set(self._input_context.copy())
```

Comment:
lets just set it on task context., we do not need to change any code, besides adding task context